networkx/algorithms/assortativity/connectivity.py

```diff
@@ -16,20 +16,25 @@ def _avg_deg_conn(G, neighbors, source_degree, target_degree,
     # "k nearest neighbors, or neighbor_connectivity
     dsum = defaultdict(float)
     dnorm = defaultdict(float)
-    for n,k in source_degree(nodes).items():
-        nbrdeg = target_degree(list(neighbors(n)))
+    # hack for single nodes
+    source_nodes = source_degree(nodes)
```

Comment:
If you put this inside the ```if nodes in G:``` section it will avoid a call AND avoid exhausting nodes (if nodes is an iterator).